src/conduits/RwaOutputConduit3.sol

```diff
@@ -0,0 +1,385 @@
+    }
+
+    /*//////////////////////////////////
+               Administration
+    //////////////////////////////////*/
+
+    /**
+     * @notice Updates a contract parameter.
+     * @param what The changed parameter name. `"quitTo"`
+     * @param data The new value of the parameter.
+     */
+    function file(bytes32 what, address data) external auth {
+        if (what == "quitTo") {
+            quitTo = data;
+        } else {
+            revert("RwaOutputConduit3/unrecognised-param");
+        }
+
+        emit File(what, data);
+    }
+
+    /**
+     * @notice Sets `who` address as the recipient.
+     * @param who Recipient address.
+     * @dev `who` address should have been whitelisted using `kiss`.
+     */
+    function pick(address who) external {
+        require(can[msg.sender] == 1, "RwaOutputConduit3/not-operator");
+        require(bud[who] == 1 || who == address(0), "RwaOutputConduit3/not-bud");
+        to = who;
+        emit Pick(who);
+    }
+
+    /*//////////////////////////////////
+               Operations
+    //////////////////////////////////*/
+
+    /**
+     * @notice Method to swap DAI contract balance to GEM through PSM and push it to the recipient address.
+     * @dev `msg.sender` must have been `mate`d and `to` must have been `pick`ed.
+     */
+    function push() external isMate {
+        _doPush(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Swaps the specified amount of DAI into GEM through the PSM and push it to the recipient address.
+     * @dev `msg.sender` must have been `mate`d and `to` must have been `pick`ed.
+     * @param wad DAI amount.
+     */
+    function push(uint256 wad) external isMate {
+        _doPush(wad);
+    }
+
+    /**
+     * @notice Flushes out any DAI balance to `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     */
+    function quit() external isMate {
+        _doQuit(dai.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Flushes out the specified amount of DAI to the `quitTo` address.
+     * @dev `msg.sender` must have received push access through `mate()`.
+     * @param wad DAI amount.
+     */
+    function quit(uint256 wad) external isMate {
+        _doQuit(wad);
+    }
+
+    /**
+     * @notice Flushes out any `token` balance to `usr` address.
+     * @param token Destination address.
+     * @param usr Destination address.
+     * @param amt Token amount.
+     * @dev Can be called only by admin.
+     */
+    function yank(
+        address token,
+        address usr,
+        uint256 amt
+    ) external auth {
+        GemAbstract(token).transfer(usr, amt);
+        emit Yank(token, usr, amt);
+    }
+
+    /**
+     * @notice Calculate expected amount of GEM received for swapping `wad` amount of DAI.
+     * @param wad DAI amount.
+     * @return gemAmt Amount of GEM expected.
+     */
+    function expectedGemAmt(uint256 wad) public view returns (uint256) {
```

Comment:
mm is this really the public helper function we need? it is more of a validation function for the input wad amount.
I would think a more useful function is something like `requiredDai`, which would tell governance and the mate how much dai needs to be drawn and  swapped to get the exact usdc amount (as the psm fees stay in maker hands anyway, and I assume the deal is denominated in usdc amounts).

I guess there's not a lot of harm in leaving `expectedGemAmt`public, but if it's not the helper function we need then I'd treat it the same as its parralel `_gemToDai` (both either public or internal/inlined).

If it was me, and assuming I'm not wrong about the deal terms and the psm fee usage, I'd probably:
* make an external `requiredDai` function.
* remove `requiredGemAmt` (as suggested here - https://github.com/makerdao/mip21-toolkit/pull/21#discussion_r973705053)
* inline _gemToDai and expectedGemAmt code (for simplicity and removing interfaces).

Reply:
Sorry can you clarify what two helpres you mean and whether you plan to change anything.

Reply:
We leave `requiredGemAmt` as public and add another method `requiredDaiAmt` which will calculate how much dai needs to be drawn and swapped to get the exact usdc amount.